From 63e15ba8d5f95b13d3abf359da718537d769f112 Mon Sep 17 00:00:00 2001
From: Tomi Valkeinen <tomi.valkeinen@nokia.com>
Date: Tue, 7 Apr 2009 10:01:58 +0300
Subject: [PATCH] DSS2: check for ovl paddr only when enabling

It seems Xvideo uses SETUP_PLANE ioctl even when
the fb memory has not been allocated. Sigh.
---
 drivers/video/omap2/dss/overlay.c |    8 +++++---
 1 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/drivers/video/omap2/dss/overlay.c b/drivers/video/omap2/dss/overlay.c
index 9209acf..c047206 100644
--- a/drivers/video/omap2/dss/overlay.c
+++ b/drivers/video/omap2/dss/overlay.c
@@ -281,6 +281,11 @@ int dss_check_overlay(struct omap_overlay *ovl, struct omap_display *display)
 
 	info = &ovl->info;
 
+	if (info->paddr == 0) {
+		DSSDBG("check_overlay failed: paddr 0\n");
+		return -EINVAL;
+	}
+
 	display->get_resolution(display, &dw, &dh);
 
 	DSSDBG("check_overlay %d: (%d,%d %dx%d -> %dx%d) disp (%dx%d)\n",
@@ -331,9 +336,6 @@ static int dss_ovl_set_overlay_info(struct omap_overlay *ovl,
 	int r;
 	struct omap_overlay_info old_info;
 
-	if (info->paddr == 0)
-		return -EINVAL;
-
 	old_info = ovl->info;
 	ovl->info = *info;
 
-- 
1.5.6.5

